pkg/kubeapiserver/authenticator/config.go

```diff
@@ -136,6 +137,10 @@ func (config AuthenticatorConfig) New() (authenticator.Request, *spec.SecurityDe
 		authenticators = append(authenticators, serviceAccountAuth)
 		hasTokenAuth = true
 	}
+	if config.BootstrapTokenAuthenticator != nil {
```

Comment:
this also needs to be gated on a bool bound to a flag the user has to opt into (e.g. not on by default). We should mark the flag that enables this authenticator as alpha or experimental so we have freedom to change it in future releases.